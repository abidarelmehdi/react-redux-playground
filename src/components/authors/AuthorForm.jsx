import React from "react";
import { StageSpinner } from "react-spinners-kit";
import TextInput from "../core/Forms/TextInput";
import SubmitBtn from "../core/Forms/SubmitBtn";

export default function AuthorForm({ author, onChange, onSubmit, saving }) {
  return (
    <form onSubmit={onSubmit} className="py-6 px-4 sm:px-6 lg:px-8">
      <div>
        <h3 className="text-2xl leading-6 font-medium text-gray-900">
          Author Form
        </h3>
        <p className="mt-1 max-w-2xl text-sm leading-5 text-gray-500">
          Update and create new author.
        </p>
      </div>
      <div className="mt-6 sm:mt-5">
        <div className="sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5">
          <TextInput
            id="name"
            label="Name"
            name="name"
            value={author.name}
            onChange={onChange}
          />
        </div>
        <div className="mt-6 sm:mt-5 flex justify-end sm:border-t sm:border-gray-200 sm:pt-5">
          {saving ? (
            <StageSpinner size={40} color="#252f3f" />
          ) : (
            <SubmitBtn label="submit" />
          )}
        </div>
      </div>
    </form>
  );
}
