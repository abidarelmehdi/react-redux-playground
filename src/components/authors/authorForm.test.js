import React from "react";
import AuthorForm from "./AuthorForm";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import { courses, authors } from "../../../tools/mockData";

function buildAuthorForm(args) {
  const defaultProps = {
    author: authors[0],
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    saving: false,
  };

  const props = { ...defaultProps, ...args };

  return <AuthorForm {...props} />;
}

afterEach(cleanup);

// React-Testing-Library test
it("renders page title", () => {
  const { getByText } = render(buildAuthorForm());
  getByText("Author Form");
});

// Enzyme test
it("renders form", () => {
  const wrapper = shallow(buildAuthorForm());
  expect(wrapper.find("form").length).toBe(1);
});

it("renders submit btn when saving is false", () => {
  const wrapper = shallow(buildAuthorForm({ saving: false }));
  expect(wrapper.find("SubmitBtn").length).toBe(1);
});

it("renders spinner when saving is true", () => {
  const wrapper = shallow(buildAuthorForm({ saving: true }));
  expect(wrapper.find("SubmitBtn").length).toBe(0);
});

/* Snapshot tests */
it("replace save button with saving state when form is saving", () => {
  const tree = renderer.create(buildAuthorForm({ saving: true }));

  expect(tree).toMatchSnapshot();
});

/* it("load save button instead of saving state when form is loaded", () => {
  const tree = renderer.create(buildAuthorForm());

  expect(tree).toMatchSnapshot();
}) */
