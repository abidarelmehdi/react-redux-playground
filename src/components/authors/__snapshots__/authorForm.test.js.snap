// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace save button with saving state when form is saving 1`] = `
<form
  className="py-6 px-4 sm:px-6 lg:px-8"
  onSubmit={[MockFunction]}
>
  <div>
    <h3
      className="text-2xl leading-6 font-medium text-gray-900"
    >
      Author Form
    </h3>
    <p
      className="mt-1 max-w-2xl text-sm leading-5 text-gray-500"
    >
      Update and create new author.
    </p>
  </div>
  <div
    className="mt-6 sm:mt-5"
  >
    <div
      className="sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <label
        className="block text-sm font-medium leading-5 text-gray-700 sm:mt-px sm:pt-2"
        htmlFor="name"
      >
        Name
      </label>
      <div
        className="mt-1 sm:mt-0 col-span-2"
      >
        <div
          className="w-full rounded-md shadow-sm"
        >
          <input
            className="form-input block w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5"
            id="name"
            name="name"
            onChange={[MockFunction]}
            type="text"
            value="Cory House"
          />
        </div>
      </div>
    </div>
    <div
      className="mt-6 sm:mt-5 flex justify-end sm:border-t sm:border-gray-200 sm:pt-5"
    >
      <div
        className="stage__Wrapper-sc-161krao-0 iPJbLy"
        size={40}
      >
        <div
          className="stage__Ball-sc-161krao-1 VHhMl"
          color="#252f3f"
          size={40}
          x={0}
          y={16}
        />
        <div
          className="stage__Ball-sc-161krao-1 dEEzDQ"
          color="#252f3f"
          size={40}
          x={16}
          y={16}
        />
        <div
          className="stage__Ball-sc-161krao-1 jbVSnv"
          color="#252f3f"
          size={40}
          x={32}
          y={16}
        />
      </div>
    </div>
  </div>
</form>
`;
